drivers/LinstorSR.py

```diff
@@ -586,8 +586,15 @@ def create(self, uuid, size) -> None:
                 opterr='LINSTOR SR must be unique in a pool'
             )
 
-        online_hosts = util.get_online_hosts(self.session)
-        if len(online_hosts) < len(host_adresses):
+        for attempt in range(3):
+            online_hosts = util.get_online_hosts(self.session)
+            if len(online_hosts) >= len(host_adresses):
+                util.SMlog("DEBUG: All hosts online")
+                break # Ok and proceed to create
+            else:
+                util.SMlog("DEBUG: Online host: {} ; Adresses: {}".format(online_hosts, host_adresses))
+                time.sleep(15) # Sleep and retry
```

Comment:
I think it's better to modify the logic of `get_online_hosts` instead (without using host metrics, just enabled flag I guess) and not try to add a sleep call during creation.

Reply:
Can we look for reason on why host metrics were used here? Or we can introduce `get_enabled_hosts` to be at par with semantics and use it while leaving `get_online_hosts` as is. Will go with your call.

Reply:
The function `get_online_hosts` predate the git history, it won't be easy to know why

Reply:
@rushikeshjadhav I agree with you, a new function is probably a good idea. This would avoid breaking the use of the current function in cleanup.py.

Reply:
By looking at code, `online_hosts` seems necessary to fetch `hostname` and its ip address. There is verification logic about `Multiple hosts with same hostname`. 

Further it calls `_prepare_sr_on_all_hosts` with `enabled=True` which subsequently calls `_prepare_sr` that works if host is enabled. So checking for `enabled` may be redundant.

This might have to be verified with someone from xapi that if fetching ip address matters in cases of `enabled` vs `online`.  We can switch to `enabled` if ip address and hostnames are available at this stage or have to revert to `online` and wait for it.

Reply:
I think `enabled = True` should be a sufficient condition for `hostname` and `address` to be available. `Host_metrics.live` is `true` depending on connectivity with the coordinator host checked during heartbeat, so if this is important instead of just the fact of availability of the data in the database, it should be used instead.

Reply:
@last-genius Thanks. I have updated the code accordingly.